reverse_engineering/databaseService/databaseService.js

```diff
@@ -130,19 +131,32 @@ const queryDistribution = async (connectionClient, dbName, tableName, tableSchem
 	}
 };
 
-const getTableRow = async (connectionClient, dbName, tableName, tableSchema, reverseEngineeringOptions, logger) => {
+const getTableRow = async (connectionClient, dbName, tableName, tableSchema, recordSamplingSettings, logger) => {
 	const currentDbConnectionClient = await getNewConnectionClientByDb(connectionClient, dbName);
-	const percentageWord = reverseEngineeringOptions.isAbsoluteValue ? '' : 'PERCENT';
 	try {
-		return mapResponse(await currentDbConnectionClient
-			.request()
-			.input('tableName', sql.VarChar, tableName)
-			.input('tableSchema', sql.VarChar, tableSchema)
-			.input('amount', sql.Int, reverseEngineeringOptions.value)
-			.input('percent', sql.VarChar, percentageWord)
-			.query`EXEC('SELECT TOP '+ @Amount +' '+ @Percent +' * FROM [' + @TableSchema + '].[' + @TableName + '];');`);
+		let amount;
+
+		if (recordSamplingSettings.active === 'absolute') {
```

Comment:
Duplicated code in `getSampleDocSize`.
You can store all the related logic in `getSampleDocSize`, and pass getRowCount as a parameter:
```suggestion
		const getRowCount = () => getTableRowCount(tableSchema, tableName, currentDbConnectionClient);
		amount = getSampleDocSize(getRowCount, recordSamplingSettings);
```